src/script_lua.c

```diff
@@ -837,20 +837,21 @@ static robj **luaArgsToRedisArgv(lua_State *lua, int *argc) {
     return argv;
 }
 
+#define LUA_CMD_OBJCACHE_SIZE 32
+#define LUA_CMD_OBJCACHE_MAX_LEN 64
 static int luaRedisGenericCommand(lua_State *lua, int raise_error) {
-    int j;
+    int j, argc = lua_gettop(lua);
     scriptRunCtx* rctx = luaGetFromRegistry(lua, REGISTRY_RUN_CTX_NAME);
     serverAssert(rctx); /* Only supported inside script invocation */
     sds err = NULL;
     client* c = rctx->c;
     sds reply;
 
-    int argc;
-    robj **argv = luaArgsToRedisArgv(lua, &argc);
-    if (argv == NULL) {
-        return raise_error ? luaError(lua) : 1;
-    }
-
+    /* Cached across calls. */
+    static robj **argv = NULL;
+    static int argv_size = 0;
+    static robj *cached_objects[LUA_CMD_OBJCACHE_SIZE];
+    static size_t cached_objects_len[LUA_CMD_OBJCACHE_SIZE];
```

Comment:
maybe we can embed this mechanism inside luaArgsToRedisArgv?
maybe it can make the diff smaller, and also serve luaRedisAclCheckCmdPermissionsCommand.

i.e. luaArgsToRedisArgv can decide to return a static array, and we'll have a dedicated release function that will skip releasing if the argument is the static array.

Reply:
@oranagra while I solve the missing leak wanted to double check the code change was what you had in mind. Can you check it?